testapp/java/org/openyolo/testapp/SaveTestPageFragment.java

```diff
@@ -67,8 +69,16 @@ public View onCreateView(
 
     @OnClick(R.id.save_button)
     void onSave() {
-        final Credential credential = mCredentialView.makeCredentialFromFields();
-        final CredentialSaveRequest request = CredentialSaveRequest.fromCredential(credential);
+        CredentialSaveRequest request;
+        try {
+            Credential credential = mCredentialView.makeCredentialFromFields();
+            request = CredentialSaveRequest.fromCredential(credential);
+        } catch (IllegalArgumentException ex) {
```

Comment:
This might be something worth considering in our API - how to provide a convenient data validation for developers before they try to create Credential instances. But this is fine for now.